### ISSUE TITLE
Support for EF Core 6

### ISSUE THREAD

**Issue by N-Olbert** [TrackableEntities/TrackableEntities.Core#46]
Hello,

are there any plans to support EF Core 6?

Currently it seems that only EF Core 5 is fully supported, f. e. on EF Core 6 we get the following error message: 

```
System.MissingMethodException: Method not found: 'Boolean Microsoft.EntityFrameworkCore.Metadata.IForeignKey.get_IsUnique()'.
at TrackableEntities.EF.Core.NavigationExtensions.GetRelationshipType(INavigationBase navigation)
at TrackableEntities.EF.Core.DbContextExtensions.<>c.<ApplyChanges>b__0_0(EntityEntryGraphNode node)
at Microsoft.EntityFrameworkCore.ChangeTracking.ChangeTracker.<>c.<TrackGraph>b__30_0(EntityEntryGraphNode`1 n)
at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.EntityEntryGraphIterator.TraverseGraph[TState](EntityEntryGraphNode`1 node, Func`2 handleNode)
at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.EntityEntryGraphIterator.TraverseGraph[TState](EntityEntryGraphNode`1 node, Func`2 handleNode)
at Microsoft.EntityFrameworkCore.ChangeTracking.ChangeTracker.TrackGraph[TState](Object rootEntity, TState state, Func`2 callback)
at Microsoft.EntityFrameworkCore.ChangeTracking.ChangeTracker.TrackGraph(Object rootEntity, Action`1 callback)
at TrackableEntities.EF.Core.DbContextExtensions.ApplyChanges(DbContext context, ITrackable item)
at EntitiesEFCore6.DataAdapterEfCore6.SaveEntities(List`1 entities)
```

For this concrete problem the reason is the following method:
https://github.com/TrackableEntities/TrackableEntities.Core/blob/2d8ae52eb6c1b4662bc7390405a6e2e1dfaf0114/TrackableEntities.EF.Core/NavigationExtensions.cs#L16-L23

While the ForeignKey-property still [exists ](https://docs.microsoft.com/en-us/dotnet/api/microsoft.entityframeworkcore.metadata.inavigation.foreignkey?view=efcore-6.0#microsoft-entityframeworkcore-metadata-inavigation-foreignkey)within the INavigation-Interface, the IsUnique-Property does not!
Looking at the IForeigenKey-Interface documentation reveals that the IsUnique-property [was present in EF Core 5](https://docs.microsoft.com/en-us/dotnet/api/microsoft.entityframeworkcore.metadata.iforeignkey.isunique?view=efcore-5.0#microsoft-entityframeworkcore-metadata-iforeignkey-isunique), but does no longer exist [in EF Core 6](https://docs.microsoft.com/en-us/dotnet/api/microsoft.entityframeworkcore.metadata.iforeignkey?view=efcore-6.0#properties).

**Comment by tonysneed** [comment]
@N-Olbert Thank you for raising the issue. We will address as part of the .NET 6 upgrade in #45.

**Comment by samantha-letourneau** [comment]
@N-Olbert @tonysneed According to this document, .NET 6 supports .NET Standard 2.1 : https://dotnet.microsoft.com/platform/dotnet-standard. The current version used .NET Standard 2.0. 

Also, it's kind of confusing because I read that there is no version of .NET Standard that is compatible with .NET 6.0. Nevertheless, for urgent needs you can use this forked package compiled for .NET 6 : https://www.nuget.org/packages/TrackableEntities.EF.NET6/

Tony would you made an update to this repository ?

**Comment by tonysneed** [comment]
Good idea! Let me try …

**Comment by N-Olbert** [comment]
@samantha-letourneau @tonysneed  Just for reference: .NET 6 supports .NET Standard 2.1, but EF Core 6 does not:

> EF Core 6.0 requires .NET 6. EF Core 6.0 does not target any .NET Standard version; for more information see the future of .NET Standard. [Link](https://docs.microsoft.com/en-us/ef/core/what-is-new/ef-core-6.0/plan#supported-platforms)

This is also different to EF Core 5, which supported .NET Standard 2.1, as the following table indicates

| EF Core                  | 3.1 | 5.0 | 6.0 |
| -------------------- | --- | --- | --- |
| .NET & .NET Core  | 3.1 | 5.0 | 6.0|
| .NET Standard       | 2.0 | 2.1|    |

[Source](https://docs.microsoft.com/en-us/ef/core/miscellaneous/platforms#overview)

TLDR: I'm not sure if moving to .NET Standard 2.1 is enough, maybe it is required target .NET 6 directly

**Comment by tonysneed** [comment]
@N-Olbert You're right ... we'll need to port to .NET 6 for EF Core 6 compatibility. I'll get to work finalizing #45.

**Comment by tonysneed** [comment]
@N-Olbert I checked and `IsUnique` property [still exists](https://docs.microsoft.com/en-us/dotnet/api/microsoft.entityframeworkcore.metadata.ireadonlyforeignkey?f1url=%3FappId%3DDev16IDEF1%26l%3DEN-US%26k%3Dk(Microsoft.EntityFrameworkCore.Metadata.IReadOnlyForeignKey);k(DevLang-csharp)%26rd%3Dtrue&view=efcore-6.0) on `IReadOnlyForeignKey` interface.

**Comment by tonysneed** [comment]
Closed by #45. Included in release [v6.0.0](https://github.com/TrackableEntities/TrackableEntities.Core/releases/tag/v6.0.0).
